### DISCUSSION TITLE
Available GPIOs for user buttons

### DISCUSSION THREAD

**Discussion by miotislucifugis** [probonopd/MiniDexed#688] [General]
Ive designed a new pcb pi shield (or "hat", whatever) for minidexed that has footprints for some user-defined buttons.  I cant seem to find this info now, tho' I know Ive read it here somewhere before.    Before I order PCbs,  can anyone confirm that GPIOs  #0, 1, 5, 6, & 7 are free and safe to use.    Thanks in advance

**Comment by diyelectromusic** [comment]
I'd avoid 0 and 1 personally as they tend to be used with HATs, even thought circle/MiniDexed doesn't really care.  In some cases GPIO=0 is treated as "not used".  You might be ok - but easiest thing is to just try it with a jumper lead...

5,6,7 should be fine - depending on what other hardware you're using of course.

In general terms, everything is pretty much configurable.  The limits tend to be what the Raspberry Pi is assuming - UART for MIDI, I2S audio, I2C or SPI display, etc... these are all on specific pins, so as I say, it depends what other hardware you're using.

Kevin

**Comment by miotislucifugis** [comment]
great.  Thank you for the quick reply
